addons/account/models/account_tax.py

```diff
@@ -209,18 +209,21 @@ def _compute_tax_group_id(self):
     @api.model
     def default_get(self, fields_list):
         # company_id is added so that we are sure to fetch a default value from it to use in repartition lines, below
-        rslt = super(AccountTax, self).default_get(fields_list + ['company_id'])
-
-        company_id = rslt.get('company_id')
-
-        repartition = rslt.setdefault('repartition_line_ids', [])
-        if 'repartition_line_ids' in fields_list and not repartition:
-            repartition.extend([
+        if 'company_id' not in fields_list and (
+            set(['refund_repartition_line_ids', 'invoice_repartition_line_ids'])
+            & set(fields_list)
+        ):
+            fields_list += ['company_id']
+        rslt = super(AccountTax, self).default_get(fields_list)
```

Comment:
Nitpicking, fell free to ignore

* literal-set over `set([...])`
* `not set().isdisjoint(iterable)` is faster and uses less memory than `set() & set()`
* if I had to change a single thing in Python, I would remove `list.__iadd__`, this function really is misleading, I always prefer append/extend
* py3 super
```suggestion
        if 'company_id' not in fields_list and (
            not {'refund_repartition_line_ids', 'invoice_repartition_line_ids'}.isdisjoint(fields_list)
        ):
            fields_list.append('company_id')
        rslt = super().default_get(fields_list)
```